R/type-list-of.R

```diff
@@ -188,11 +188,26 @@ vec_ptype2.vctrs_list_of <- function(x, y, ..., x_arg = "", y_arg = "") {
 }
```

Comment:
If you do this and close that issue, it would be great to add a test like this to ensure `[<-` and `vec_assign()` are more in sync

``` r
library(vctrs)

x <- list_of(1L, .ptype = integer())

`[<-`(x, 1, list(2L))
#> <list_of<integer>[1]>
#> [[1]]
#> [1] 2
vec_assign(x, 1, list(2L))
#> <list_of<integer>[1]>
#> [[1]]
#> [1] 2
```

Reply:
Let's do this in another PR.